core/src/main/java/com/scalar/db/common/error/ScalarDbError.java

```diff
@@ -0,0 +1,76 @@
+package com.scalar.db.common.error;
+
+import java.util.Objects;
+
+public interface ScalarDbError {
```

Comment:
The error consists of a component name, category, sub-category, code, message, cause, and solution.